### PR TITLE
ci: add lint in cb

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

Add missing linter step.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by awsluja — APPROVED** [review]

File: codebuild_specs/pr_workflow.yml [hunk 1]
```diff
@@ -22,6 +22,10 @@ batch:
       buildspec: codebuild_specs/test.yml
       env:
         compute-type: BUILD_GENERAL1_LARGE
+    - identifier: lint
+      buildspec: codebuild_specs/lint.yml
+      env:
+        compute-type: BUILD_GENERAL1_LARGE
```
Comment (awsluja) [inline codebuild_specs/pr_workflow.yml:28]: can you add to the other workflows

Comment (sobolk) [reply codebuild_specs/pr_workflow.yml:28]: This one is needed only in PR builds see https://github.com/aws-amplify/amplify-cli/blob/392f0e2c342408b914aa1802d3520191ed70c9fd/.circleci/config.base.yml#L1064-L1068 .

Comment (sobolk) [reply codebuild_specs/pr_workflow.yml:28]: but it's included in `dev` build... so maybe it's worth adding.

**Review by awsluja — DISMISSED** [review]

File: codebuild_specs/lint.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
 version: 0.2
 env:
   shell: bash
+  variables:
+    NODE_OPTIONS: --max-old-space-size=5120
```
Comment (awsluja) [inline codebuild_specs/lint.yml:5]: good catch

**Review by awsluja — DISMISSED** [review]

**Review by awsluja — DISMISSED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**